### PR TITLE
Upload coverage from CircleCI

### PR DESCRIPTION
**Pull request by astrofrog**
### Description

Attempt to upload coverage from CircleCI

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by Cadair** [comment]
Why did circle not run on this lol

**Comment by pllim** [comment]
Is that config actually correct? I have seen jobs missing when the config is misconfigured.

**Comment by astrofrog** [comment]
Just in case CircleCI does not run on drafts, marking this as ready for review but not actually ready

**Comment by pllim** [comment]
Nice try but I am pretty sure CircleCI runs on draft PRs. 😉

**Comment by astrofrog** [comment]
I don't understand what is going on, as the CircleCI CLI validates my config...

**Comment by pllim** [comment]
Is there some special permission or settings we need to change either in this repo or over at CircleCI or codecov ?

**Comment by pllim** [comment]
What about https://github.com/astropy/astropy/blob/main/codecov.yml ? Anything we need to add there?

**Comment by pllim** [comment]
This is what happens on Actions

https://github.com/astropy/astropy/blob/0c37a7141c6c21c52ce054f2d895f6f6eacbf24b/.github/workflows/ci_workflows.yml#L108-L113

https://github.com/astropy/astropy/blob/0c37a7141c6c21c52ce054f2d895f6f6eacbf24b/tox.ini#L26

https://github.com/astropy/astropy/blob/0c37a7141c6c21c52ce054f2d895f6f6eacbf24b/tox.ini#L124

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 version: 2.1
 
+orbs:
+  codecov: codecov/codecov@3
```
Comment (pllim) [inline .circleci/config.yml:4]: What repo does this point to? I cannot find https://github.com/codecov/codecov , only only https://github.com/codecov/codecov-circleci-orb

Comment (pllim) [reply .circleci/config.yml:4]: Oh nvm, I see it https://circleci.com/developer/orbs/orb/codecov/codecov

Comment (pllim) [reply .circleci/config.yml:4]: I guess another thing to check is whether this is available for our tier or you have to pay extra... 🤷

**Comment by pllim** [comment]
Yay, they appeared! But what's up with this?

```
__________________________ TestBasic.test_tick_angles __________________________
Hash 49ac60171ec7d3c7e4458d429529326b1fd12963d7199725cfdda1ad39c213b3 doesn't match hash 7210886e657160823c2ce6d02f17ffa1722cdb340d82e74b58c3e5d220bc1960 in library /home/circleci/project/astropy/tests/figures/py39-test-image-mpl311-cov.json for test astropy.visualization.wcsaxes.tests.test_images.TestBasic.test_tick_angles.

Image comparison test
---------------------
Could not download baseline image from any of the available URLs
__________________ TestBasic.test_tick_angles_non_square_axes __________________
Hash 0b1c86bb02ccaf595e7a6b98f4dfa8046a1b95d6fd7e07508b5bb0164a287eba doesn't match hash b023ffeccd111207ff9fea35f8ed6606a2b7f3367ef053f5957ee5c25561fb44 in library /home/circleci/project/astropy/tests/figures/py39-test-image-mpl311-cov.json for test astropy.visualization.wcsaxes.tests.test_images.TestBasic.test_tick_angles_non_square_axes.

Image comparison test
---------------------
Could not download baseline image from any of the available URLs
```

**Comment by astrofrog** [comment]
@Cadair @ConorMacBride - any idea why the hashes would change for a couple of figures just from adding -cov to the tox env?

**Comment by astrofrog** [comment]
Ok so the RMS difference is 0.0033 so I think safe to ignore - not really sure what's going on but I can just update the hashes.

**Comment by astrofrog** [comment]
I don't understand codecov - it seems to suggest the coverage has decreased even though when you look at the details no files have decreased coverage: https://app.codecov.io/gh/astropy/astropy/pull/13928 - and some have much better coverage 🤯 Anyone understand codecov maths?

**Comment by ConorMacBride** [comment]
> @Cadair @ConorMacBride - any idea why the hashes would change for a couple of figures just from adding -cov to the tox env?

Only thing I can think of is maybe the reduced optimisation produces slightly different values in some cases? https://github.com/astropy/astropy/blob/2863c9da53ba8e7d9e534c9e312b6a2f04ab62dd/tox.ini#L26 

I notice the codecov total being a bit off quite often. Always assumed it was just slow to update the total after it processes each report, but it's been a while!

**Review by Cadair — APPROVED** [review]
Looks like it works :+1:

**Comment by jdavies-st** [comment]
If you have one CI job that does one subset of tests and another that does some other subset, then you'll want to use codecov flags to label these.  That way they can be compared to like "flags" and one subset won't be compared to another.

**Comment by Cadair** [comment]
Interesting, I have not used that feature before. Could you open an issue about it?

**Comment by pllim** [comment]
Issue opened at https://github.com/astropy/astropy/issues/13936

**Pull request merged**